cpp/src/arrow/csv/parser_test.cc

```diff
@@ -376,6 +383,19 @@ TEST(BlockParser, TruncatedData) {
   }
 }
 
+TEST(BlockParser, TruncatedDataViews) {
+  // If non-last block is truncated, parsing stops
+  BlockParser parser(ParseOptions::Defaults(), /*num_cols=*/3);
+  AssertParsePartial(parser, Views({"a,b,", "c\n"}), 0);
+  // (XXX should we guarantee this one below?)
```

Comment:
That's a good point, let me try that.

Reply:
Ok, so the problem is that it makes harder to return a meaningful error message. The CSV parser itself has no idea of who did the chunking and how, so it can't start mentioning `newlines_in_values`. It would have to return a more generic `Status::Invalid`, then the CSV would have to detect that particular `Status::Invalid` (how?) and rewrite the error message.

This is a lot of complication for a bit of unnecessary purity in the `BlockParser` API (which isn't terribly pretty to begin with), so I would rather keep the current solution in the PR.

Reply:
I don't think the BlockParser actually needs to return an error (*maybe* a debug assertion). I'm mostly concerned with how we test your solution. By analogy: if we had a calculator object, it'd be better to test whatever uses that object rather than unit testing `calculator.divide(1, 0)`. BlockParser might be refactored later in a way which doesn't change reader behavior as long as chunking is valid but *does* invalidate these unit tests, and I'd like to avoid placing such a constraint on future developers.

Reply:
Nit: Could we at least add a comment indicating that these tests exercise a condition which would be invalid for BlockParser?

Reply:
Yes, I'll add it.